src/middlewares/validators/druidValidator.ts

```diff
@@ -0,0 +1,69 @@
+import Ajv from "ajv";
+import fs from "fs";
+import { NextFunction, Request, Response } from "express";
+import { IValidationResponse } from "../../models";
+import { ValidationService } from "../../services/validationService";
+import { config } from "../../configs/config";
+import createError from "http-errors";
+import httpStatus from "http-status";
+
+const limits = config.limits;
+let requestBodySchema = JSON.parse(
+  fs.readFileSync(process.cwd() + "/src/configs/requestBodySchema.json", "utf8")
+);
+
+const validate =
+  (configParams: Record<string, any>) =>
+  (req: Request, res: Response, next: NextFunction) => {
+    const queryPayload = req.body;
+    let ajv = new Ajv();
+    let validRequestObj = ajv.validate(
+      requestBodySchema,
+      JSON.parse(JSON.stringify(queryPayload))
+    );
+    if (validRequestObj) {
+      const isSqlQuery = configParams.isSqlQuery;
+      let dataSource, result: IValidationResponse;
+      if (isSqlQuery) {
+        queryPayload.querySql.query = queryPayload.querySql.query
+          .replace(/\s+/g, " ")
+          .trim();
```

Comment:
can we create a function to validate both SQL, and native queries? it will be easy to add other query types in later days.